sdk/core/switcher/entry.S

```diff
@@ -1659,6 +1659,7 @@ exception_entry_asm:
 	 *  mtdc: pointer to TrustedStack
 	 */
 	csrw               mcause, MCAUSE_THREAD_EXIT
+	csrw               mtval, MCAUSE_THREAD_EXIT
```

Comment:
Worth a comment here to indicate that this avoids a small info leak to the scheduler?